File: python/src/interface/catalogo.py
# interface/catalogo.py


from utils import limpar_tela, continuar, opcao_errada
from interface.menus import menu_livro_disponivel, menu_livro_emprestado


def mostrar_catalogo(catalogo):
    limpar_tela()
    print("-------------- Catálogo --------------\n")

    for idx, livro in enumerate(catalogo, start=1):
        print(f"Livro - {idx}")
        print(f"Título: {livro.titulo}")
        print(f"Autor: {livro.autor}")
        print(f"Ano: {livro.ano_publicacao}")
        print(f"Género: {livro.genero}")
        print(f"Avaliação Média: {livro.mediaAvaliacoes()}\n")


def interagir_com_livro(catalogo, indice):
    livro = catalogo[indice - 1]

    while True:
        limpar_tela()
        print(f"Título: {livro.titulo}")
        print(f"Avaliação Média: {livro.mediaAvaliacoes()}\n")

        if not livro.emprestado:
            menu_livro_disponivel()
        else:
            menu_livro_emprestado()

        opcao = input("\nEscolha uma opção: ")

        if opcao == "1":
            nome = input("Digite o seu nome: ")
            if livro.emprestado:
                livro.devolver()
                print("Livro devolvido com sucesso!")
            else:
                livro.emprestar()
                print("Livro requisitado com sucesso!")
            continuar()

        elif opcao == "2":
            while True:
                try:
                    nota = int(input("Avaliação (1 a 10): "))
                    if 1 <= nota <= 10:
                        livro.adicionarAvaliacao(nota)
                        print("Avaliação registada!")
                        continuar()
                        break
                    else:
                        opcao_errada()
                except ValueError:
                    opcao_errada()

        elif opcao == "0":
            break
        else:
            opcao_errada()

File: javascript/interface/catalogo.js
// interface/catalogo.js


const { menuLivroDisponivel, menuLivroEmprestado } = require('./menus');

function mostrarCatalogo(catalogo) {
    console.clear();
    console.log("-------------- Catálogo --------------\n");
    catalogo.forEach((livro, index) => {
        console.log(`Livro - ${index + 1}`);
        console.log(`Título: ${livro.titulo}`);
        console.log(`Autor: ${livro.autor}`);
        console.log(`Ano: ${livro.anoPublicacao}`);
        console.log(`Gênero: ${livro.genero}`);
        console.log(`Avaliação Média: ${livro.mediaAvaliacoes()}\n`);
    });
}

const prompt = require('prompt-sync')({sigint: true}); // npm install prompt-sync

function interagirComLivro(catalogo, indice) {
    const livro = catalogo[indice - 1];

    while (true) {
        console.clear();
        console.log(`Título: ${livro.titulo}`);
        console.log(`Avaliação Média: ${livro.mediaAvaliacoes()}\n`);

        if (!livro.emprestado) {
            menuLivroDisponivel();
        } else {
            menuLivroEmprestado();
        }

        const opcao = prompt("\nEscolha uma opção: ");

        if (opcao === "1") {
            const nome = prompt("Digite o seu nome: ");
            if (livro.emprestado) {
                livro.devolver();
                console.log("Livro devolvido com sucesso!");
            } else {
                livro.emprestar();
                console.log("Livro requisitado com sucesso!");
            }
            prompt("Pressione ENTER para continuar...");
        } else if (opcao === "2") {
            while (true) {
                try {
                    const nota = parseInt(prompt("Avaliação (1 a 10): "));
                    if (nota >= 1 && nota <= 10) {
                        livro.adicionarAvaliacao(nota);
                        console.log("Avaliação registada!");
                        prompt("Pressione ENTER para continuar...");
                        break;
                    } else {
                        console.log("Opção inválida!");
                    }
                } catch {
                    console.log("Opção inválida!");
                }
            }
        } else if (opcao === "0") {
            break;
        } else {
            console.log("Opção inválida!");
        }
    }
}

module.exports = { mostrarCatalogo, interagirComLivro };
